### PR TITLE
[feat] Added support for Ubuntu 20.04 (#189)

### LINKED ISSUE #189: [OS] Add support for Ubuntu 20 LTS
Add support for Ubuntu 20 LTS.

### PR DESCRIPTION
**Pull request by PabloCastellano**
(this PR integrates the work done in #190)

- switched to use /usr/bin/python3 by default
- python-virtualenv > python3-virtualenv
- build in travis using ubuntu 20.04
- add ubuntu 20.04 (focal) to meta info
- handle redis 4/5 and dependencies
- removed step to install python-wheel (installed by pip later)
- simplified python3-virtualenv installation
- pin channels_redis to 2.4 for redis 4 compatibility

Closes #189

### CODE REVIEW

File: files/generate_django_secret_key.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (nemesisdesign) [inline files/generate_django_secret_key.py:1]: are you sure this is needed? The virtualenv should know the python version without the need of introducing this change

**Review by nemesisdesign — COMMENTED** [review]
Looks good, only one minor doubt below.

**Comment by KapilBansal320** [comment]
@PabloCastellano the type should be feature instead of feat

**Comment by nemesisdesign** [comment]
> @PabloCastellano the type should be feature instead of feat

More info here: https://github.com/openwisp/openwisp-utils/issues/110#issue-620578722

**Review by nemesisdesign — APPROVED** [review]
Fixed the minor issue https://github.com/openwisp/ansible-openwisp2/pull/207/commits/26654f0ec512f868599fd926a0f721b644058414.

Thanks!

File: files/generate_django_secret_key.py (hunk 1 above)
Comment (PabloCastellano) [reply files/generate_django_secret_key.py]: Not sure now, but I probably made this change and then forgot to remove it :grin:

**Pull request merged**